src/api/app/controllers/person_controller.rb

```diff
@@ -193,16 +222,21 @@ def internal_register
   end
 
   def update_watchlist(user, xml)
-    new_watchlist = []
-    xml.get('watchlist').elements('project') do |e|
-      new_watchlist << e['name']
-    end
+    # Keeping the backwards compatibility
+    if Flipper.enabled?(:new_watchlist, user)
```

Comment:
In general, everything I tested manually worked fine when the user performing the action is the same as the user to be modified (the one we are editing in the XML).

However, when an admin user is trying to see or update another user, this is not working as expected.

The reason is that in ` check_user_belongs_feature_flag` we are checking `Flipper.enabled?(:new_watchlist, User.session)` so the user performing the action (_Admin_), however, in the line I'm commenting, we are checking `Flipper.enabled?(:new_watchlist, user)` where `user` is not the one in the session but the one given by the `login` tag in the XML (_scp_).